src/updater.cpp

```diff
@@ -19,1100 +15,663 @@
-            zsync2::trim(newSignature);
+    bool Updater::remoteFileSize(off_t& fileSize) const {
+        // only available update method is via ZSync
+        if (d->zSyncClient != nullptr)
+            return d->zSyncClient->remoteFileSize(fileSize);
 
-            auto oldSigned = !oldSignature.empty();
-            auto newSigned = !newSignature.empty();
+        return false;
+    }
 
-            auto oldDigest = d->hashAppImage(pathToOldAppImage);
-            auto newDigest = d->hashAppImage(pathToNewAppImage);
+    Updater::ValidationState Updater::validateSignature() {
+        std::string pathToNewAppImage;
+        if (!this->pathToNewFile(pathToNewAppImage)) {
+            // return generic error
+            return VALIDATION_FAILED;
+        }
 
-            auto oldDigestOrig = readElfSection(pathToOldAppImage, ".sha256_sig");
-            auto newDigestOrig = readElfSection(pathToNewAppImage, ".sha256_sig");
+        AppImage newAppImage(pathToNewAppImage);
 
-            std::string tempDir;
-            {
-                char* buffer;
+        auto pathToOldAppImage = abspath(d->appImage.path());
+        if (pathToOldAppImage == pathToNewAppImage) {
+            pathToOldAppImage = pathToNewAppImage + ".zs-old";
+        }
 
-                char* pattern = strdup("/tmp/AppImageUpdate-XXXXXX");
-                if ((buffer = mkdtemp(pattern)) == nullptr) {
-                    d->issueStatusMessage("Failed to create temporary directory");
-                    return VALIDATION_TEMPDIR_CREATION_FAILED;
-                }
+        AppImage oldAppImage(pathToOldAppImage);
 
-                tempDir = buffer;
+        auto oldSignature = oldAppImage.readSignature();
+        auto newSignature = newAppImage.readSignature();
 
-                free(pattern);
-            }
+        // remove any spaces and/or newline characters there might be on the left or right of the payload
+        zsync2::trim(oldSignature, '\n');
+        zsync2::trim(newSignature, '\n');
+        zsync2::trim(oldSignature);
+        zsync2::trim(newSignature);
 
-            auto tempFile = [&tempDir](const std::string& filename, const std::string& contents) {
-                std::stringstream path;
-                path << tempDir << "/" << filename;
+        auto oldSigned = !oldSignature.empty();
+        auto newSigned = !newSignature.empty();
 
-                auto x = path.str();
+        auto oldDigest = oldAppImage.calculateHash();
+        auto newDigest = newAppImage.calculateHash();
 
-                std::ofstream ofs(path.str());
-                ofs.write(contents.c_str(), contents.size());
+        auto oldDigestOrig = readElfSection(pathToOldAppImage, ".sha256_sig");
+        auto newDigestOrig = readElfSection(pathToNewAppImage, ".sha256_sig");
 
-                return path.str();
-            };
+        std::string tempDir;
+        {
+            auto buffer = makeBuffer("/tmp/AppImageUpdate-XXXXXX");
```

Comment:
Any special reason for using this instead of `char dir_path[] = {"/tmp/AppImageUpdate-XXXXXX\0"};`

Reply:
Not really.